Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -508,10 +510,84 @@ extension BrowserViewController: TopToolbarDelegate {
         self.present(container, animated: true)
       }
     }
+    
+    shields.showSubmitReportView = { [weak self] shieldsViewController in
+      shieldsViewController.dismiss(animated: true) {
+        guard let url = shieldsViewController.tab.url else { return }
+        self?.showSubmitReportView(for: url, isPrivateMode: shields.tab.isPrivate)
+      }
+    }
+    
     let container = PopoverNavigationController(rootViewController: shields)
     let popover = PopoverController(contentController: container, contentSizeBehavior: .preferredContentSize)
     popover.present(from: topToolbar.locationView.shieldsButton, on: self)
   }
+  
+  private func showSubmitReportView(for url: URL, isPrivateMode: Bool) {
+    // Strip fragments and query params from url
+    var components = URLComponents(url: url, resolvingAgainstBaseURL: false)
+    components?.fragment = nil
+    components?.queryItems = nil
+    guard let cleanedURL = components?.url else { return }
+    
+    let viewController = UIHostingController(rootView: SubmitReportView(
+      url: cleanedURL,
+      dismiss: { [weak self] in
+        self?.navigationController?.dismiss(animated: true)
+      },
+      submit: { [weak self] url, additionalDetails, contactInfo in
+        guard let self = self else { return }
+        self.navigationController?.dismiss(animated: true)
+        let popover = PopoverController(
+          contentController: SubmitReportSuccessViewController(), 
+          contentSizeBehavior: .preferredContentSize
+        )
+        
+        let domain = Domain.getOrCreate(forUrl: url, persistent: !isPrivateMode)
+        
+        let report = WebcompatReporter.Report(
+          cleanedURL: url,
+          additionalDetails: additionalDetails,
+          contactInfo: contactInfo,
+          areShieldsEnabled: !domain.areAllShieldsOff,
+          adBlockLevel: domain.blockAdsAndTrackingLevel,
+          fingerprintProtectionLevel: domain.finterprintProtectionLevel,
+          adBlockListTitles: FilterListStorage.shared.filterLists.compactMap({ filterList -> String? in
+            guard filterList.isEnabled else { return nil }
+            return filterList.entry.title
+          }),
+          isVPNEnabled: BraveVPN.isConnected
+        )
+        
+        Task { @MainActor in
+          await WebcompatReporter.send(report: report)
+          try await Task.sleep(seconds: 4)
+          
+          if !popover.isBeingDismissed {
+            self.navigationController?.dismiss(animated: true)
+          }
+        }
```

Comment:
This is still a problem with keeping self alive for 4 seconds even if the view is dismissed earlier

Reply:
Sorry yah I'll try to use a timer. I thought since its on the BVC (which is alive for the lifetime of the app) its not as big a deal and avoids extra code

Reply:
I'd recommend even moving this logic into the View itself, make the "submit" action actually just an async closure meant only to build and submit the report. That would also let you add a loader to the submit button since you're actually waiting for the call to finish before moving to show the success

Reply:
Done

Reply:
Ok, I moved it to the view